plugins/callback/scalyr.py

```diff
@@ -0,0 +1,234 @@
+  - 'Based on the splunk and sumologic callbacks'
+options:
+  url:
+    description: URL to the Scalyr API endpoint
+    default: https://app.eu.scalyr.com/api/addEvents
+    env:
+      - name: SCALYR_URL
+    ini:
+      - section: callback_scalyr
+        key: url
+    type: string
+  authtoken:
+    description: Token to authenticate the connection to Scalyr.
+    env:
+      - name: SCALYR_AUTHTOKEN
+    ini:
+      - section: callback_scalyr
+        key: authtoken
+    type: string
+'''
+
+EXAMPLES = '''
+examples: >
+  To enable, add this to your ansible.cfg file in the defaults block
+    [defaults]
+    callback_whitelist = community.general.scalyr
+
+  Set the environment variable
+    export SCALYR_URL=https://app.eu.scalyr.com/api/addEvents
+    export SCALYR_AUTHTOKEN=XXXXXXXXXXXXXXXXXXXXXmlQ0-
+
+  Set the ansible.cfg variable in the callback_scalyr block
+    [callback_scalyr]
+    url = https://app.eu.scalyr.com/api/addEvents
+    authtoken = XXXXXXXXXXXXXXXXXXXXXmlQ0-
+'''
+
+import getpass
+import json
+import socket
+import time
+import uuid
+
+from datetime import datetime
+from os.path import basename
+
+from ansible.module_utils.urls import open_url
+from ansible.module_utils.ansible_release import __version__ as ansible_version
+from ansible.parsing.ajson import AnsibleJSONEncoder
+from ansible.plugins.callback import CallbackBase
+
+
+def secondsToNanoSeconds(seconds):
+    return seconds * 1000000000
+
+class ScalyrLogSource(object):
+    def __init__(self):
+        self.ansible_check_mode = False
+        self.ansible_playbook = ""
+        self.session = str(uuid.uuid4())
+        self.host = socket.gethostname()
+        self.ip_address = socket.gethostbyname(socket.gethostname())
+        self.user = getpass.getuser()
+
+    def send_event(self, url, authtoken, state, result, runtime):
+        if result._task_fields['args'].get('_ansible_check_mode') is True:
+            self.ansible_check_mode = True
+
+        if result._task._role:
+            ansible_role = str(result._task._role)
+        else:
+            ansible_role = None
+
+        if 'args' in result._task_fields:
+            del result._task_fields['args']
+
+        data = {
+            "token": str(authtoken),
+            "session": str(self.session),
+            "sessionInfo": {
+                "serverHost": self.host,
+                "serverIpAddress": self.ip_address,
+                "logfile": "ansible.log",
+                "parser": ""
+            },
+            "events": [
+                {
+                    "ts": str(int(secondsToNanoSeconds(time.time()))) ,
+                    "type": 0,
+                    "sev": 3,
+                    "attrs": {
+                        "uuid": result._task._uuid,
+                        "status": state,
+                        "runtime": runtime,
+                        "user": self.user,
+                        "ansible_version": ansible_version,
```

Comment:
Please note that this is not the Ansible version, but the ansible-base / ansible-core version.

Reply:
Is there any other way to get the ansible version=

Reply:
There's no way, actually. You can only get hold of the ansible-base/ansible-core version, which happens to be the Ansible version for Ansible 2.9.